### DISCUSSION TITLE
How to change behavior of step control buttons

### DISCUSSION THREAD

**Discussion by maxwellsmart84** [vueform/vueform#159] [Questions] [answered]
Hey there - I have a form and on the last step we have "finish".  I would like to do two things:

1. Change the button text to "Close" after the form has been submitted
2. Change the behavior of the finish event to close the modal I have my form step component in.

I have opened some discussions in regards to the behavior or controlling these buttons but can't seem to change the behavior on step by step basis.  The only workaround I have right now is completely hide the Vueform element and render a div with the content I want.

Is anyone else getting this to work?  I have tried using the `:label="{ finish: template: {.... } }"` but this doesn't work at all. I can open another sandbox if that is helpful demonstrating this not working, however I have created 2 other discussions that show there are some potential bugs surrounding the buttons in FormStepControl.

**Comment by adamberecz** [comment, marked as answer]
So basically you would need to disable the "submit" action on the Finish button after the first submit. That would be tricky, but not impossible with a plugin.

However a better way might be to just hide the form controls after submit and render a simple button that closes the window.

Here's one way of doing that:
https://stackblitz.com/edit/github-1moftk?file=src%2FApp.vue

```vue
<script setup>
import { ref } from 'vue';

const submitted = ref(false);

const handleSubmit = () => {
  submitted.value = true;
};

const handleClose = () => {
  alert('Closing modal');
};
</script>

<template>
  <div class="max-w-[500px] mx-auto mt-10">
    <Vueform @submit="handleSubmit">
      <template #empty>
        <FormSteps>
          <FormStep
            :buttons="{
              previous: false,
              finish: false,
            }"
            name="test"
            label="Test"
            :elements="['hello_world', 'close']"
          />
        </FormSteps>
        <FormElements>
          <TextElement name="hello_world" label="Hello" placeholder="World" />
          <ButtonElement
            name="close"
            :conditions="[() => submitted]"
            @click="handleClose"
          >
            Close
          </ButtonElement>
        </FormElements>
        <FormStepsControls v-show="!submitted" />
      </template>
    </Vueform>
  </div>
</template>

<style></style>
```

Pressing the Finish button is essentially submitting the form, so that's why `@submit` works in this case.
